nodes/modifier_make/isoLine2D.py

```diff
@@ -0,0 +1,694 @@
+
+        edgesOut.append(ed)
+        
+        edIndexOld=edIndex
+        vIndex=e[1]
+        
+        
+        for k in range(len(edgesCopy)):
+            
+            if vIndex == edgesCopy[k][1][0] :
+                edIndex=edgesCopy[k][0]
+                edgesCopy.pop(k)
+                orSide =0
+                break
+            elif vIndex== edgesCopy[k][1][1] :
+                    edIndex=edgesCopy[k][0]
+                    edgesCopy.pop(k)
+                    orSide =1
+                    break
+        if edIndex == edIndexOld and len(edgesCopy)>0:
+            edIndex=edgesCopy[0][0]
+
+            orSide =0
+            edgesCopy.pop(0)
+
+                    
+    return vertsOut, edgesOut
+
+                        
+def buildNet(verts_in, edges_in,vLen,Radius):
+    net=[]
+    for j in range(vLen):
+        connect=[]
+        net.append(connect)
+    for ed in edges_in:
+        
+        s0 = ed[0]
+        s1 = ed[1]
+        if s0 > len(verts_in)-1 or s1 > len(verts_in)-1:
+            break
+        v0 = verts_in[s0]
+        v1 = verts_in[s1]
+        an = atan2(v1[1]-v0[1],v1[0]-v0[0]) + pi
+        an2 = atan2(v0[1]-v1[1],v0[0]-v1[0]) + pi
+        r1 = Radius[s0%len(Radius)]
+        r2 = Radius[s1%len(Radius)]
+        dist = pow(pow(v1[0]-v0[0], 2) + pow(v1[1]-v0[1], 2), 0.5)
+        beta = asin( min(max((r1-r2)/dist, -1), 1) )
+        
+        net[s0].append((an + 0.5*pi) % (2*pi))
+        net[s0].append((an + 0.5*pi + beta) % (2*pi))
+        net[s0].append((an + 1.5*pi - beta)%(2*pi))
+        
+        net[s1].append((an2 + 0.5*pi) % (2*pi))
+        net[s1].append((an2 + 0.5*pi - beta) % (2*pi))
+        net[s1].append((an2 + 1.5*pi + beta) % (2*pi))
+
+    return net
+    
+def listMatcher(a,listMatch):
+    if listMatch == "Long Cycle":
+        return match_long_cycle(a)
+    else:
+        return match_long_repeat(a)
+    
+class SvIsoLinesNode(bpy.types.Node, SverchCustomTreeNode):
+    ''' 2D Path with offset '''
+    bl_idname = 'SvIsoLinesNode'
+    bl_label = 'Isoline 2D'
+    bl_icon = 'MESH_CIRCLE'
+
+    modeI = EnumProperty(
+        name="modeI", description = "Constant or weigted distance when multiple radius are given",
+        items=modeItems, default="Constant", update=updateNode)
+    listMatch = EnumProperty(
+        name="listMatch", description = "Behaviour on diffent list lengths",
+        items=listMatchItems, default="Long Cycle", update=updateNode)
+    
+    rm_doubles = bpy.props.FloatProperty(
+        description = "Remove Doubles Distance",
+        min=0.0, default=0.0001, step=0.1, update=updateNode)
+    epsilon = bpy.props.FloatProperty(
+        description = "Intersection tolerance",
+        min=1.0e-5, default=1.0e-5, step=0.02, update=updateNode)
+    maskT = bpy.props.FloatProperty(
+        description = "Mask tolerance",
+        min=-1.0, default=1.0e-5, step=0.02, update=updateNode)
+    rad_ = FloatProperty(name='Radius', description='Radius',
```

Comment:
sort out the sudden weird formatting here.  dont have spaces between varname=varvalue inside a function call.